spec/immutable-data-structures.html

```diff
@@ -506,6 +506,53 @@ <h1>Tuple.prototype.filter ( _callbackfn_ [ , _thisArg_ ] )</h1>
         <h1>Tuple.prototype.find</h1>
         <p>The initial value of *Tuple.prototype.find* is  %Array.prototype.find%.</p>
       </emu-clause>
+      <emu-clause id="sec-tuple.prototype.flat">
+        <h1>Tuple.prototype.flat ( [ _depth_ ] )</h1>
+        <p>When the `flat` method is called with zero or one arguments, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Let _list_ be _T_.[[Sequence]].
+          1. Let _depthNum_ be 1.
+          1. If _depth_ is not *undefined*, then
+            1. Set _depthNum_ to ? ToInteger(_depth_).
+          1. Let _flat_ be a new empty List.
+          1. Perform ? FlattenIntoList(_flat_, _list_, _depthNum_).
+          1. Return a new Tuple value whose [[Sequence]] is _flat_.
+        </emu-alg>
+
+        <emu-clause id="sec-flattenintolist" aoid="FlattenIntoList">
+          <h1>FlattenIntoList ( _target_, _source_, _depth_ [ , _mapperFunction_, _thisArg_ ] )</h1>
```

Comment:
Nit: Maybe we should call this "FlattenIntoTuple" since the whole thing is pretty Tuple-specific, really.